app/lib/staypuft/network_query.rb

```diff
@@ -59,6 +59,23 @@ def subnet_for_host(subnet_type_name, host=@host)
       interface_hash_for_host(subnet_type_name, host)[:subnet]
     end
 
+    def interfaces_for_tenant_subnet(host=@host)
+      subnet = subnet_for_host(Staypuft::SubnetType::TENANT, host)
+      top_level_interface = host.interfaces.where(identifier:
+        interface_hash_for_host(Staypuft::SubnetType::TENANT,
+          host)[:interface]).first
```

Comment:
Also, make sure this works in cases where there is no tenant interface on the host (i.e. subnet and interface are nil) --